legion/engine/core/particles/particleemitter.cpp

```diff
@@ -1,4 +1,88 @@
-#include <core/particles/particleemitter.hpp>

namespace legion::core
{
    void particle_emitter::set_alive(size_type idx, bool alive)
    {
        m_livingBuffer[idx] = alive;
    }

    void particle_emitter::set_alive(size_type start, size_type end, bool alive)
    {
        std::fill_n(m_livingBuffer.begin() + start, end - start, alive);
    }

    bool particle_emitter::is_alive(size_type idx) const
    {
        if (idx < m_livingBuffer.size())
            return m_livingBuffer[idx];
        return false;
    }

    void particle_emitter::swap(size_type idx1, size_type idx2)
    {
        std::iter_swap(m_livingBuffer.begin() + idx1, m_livingBuffer.begin() + idx2);

        for (auto& [id, buffer] : m_particleBuffers)
            m_particleBuffers[id]->swap(idx1, idx2);
    }
+#include <core/particles/particleemitter.hpp>
+
+namespace legion::core
+{
+    void particle_emitter::play() noexcept
+    {
+        m_pause = false;
+    }
+
+    void particle_emitter::pause() noexcept
+    {
+        m_pause = true;
+    }
+
+    void particle_emitter::stop()
+    {
+        pause();
+        auto cap = capacity();
+        resize(0);
+        m_particleCount = 0;
+        resize(cap);
```

Comment:
this doesn't seem right, stop emitting shouldn't kill all particles. also not very cheap to just set everything to 0x00000000 unnecessarily